src/main/java/org/jabref/gui/openoffice/DetectOpenOfficeInstallation.java

```diff
@@ -0,0 +1,183 @@
+package org.jabref.gui.openoffice;
+
+import java.awt.BorderLayout;
+import java.io.File;
+import java.nio.file.Path;
+import java.nio.file.Paths;
+import java.util.List;
+import java.util.Optional;
+
+import javax.swing.BorderFactory;
+import javax.swing.DefaultListModel;
+import javax.swing.JDialog;
+import javax.swing.JFileChooser;
+import javax.swing.JLabel;
+import javax.swing.JList;
+import javax.swing.JOptionPane;
+import javax.swing.JProgressBar;
+import javax.swing.ListSelectionModel;
+import javax.swing.SwingConstants;
+
+import org.jabref.gui.worker.AbstractWorker;
+import org.jabref.logic.l10n.Localization;
+import org.jabref.logic.openoffice.OpenOfficeFileSearch;
+import org.jabref.logic.openoffice.OpenOfficePreferences;
+import org.jabref.logic.util.OS;
+import org.jabref.logic.util.io.FileUtil;
+
+import com.jgoodies.forms.builder.FormBuilder;
+import com.jgoodies.forms.layout.FormLayout;
+
+/**
+ * Tools for automatically detecting OpenOffice or LibreOffice installations.
+ */
+public class DetectOpenOfficeInstallation extends AbstractWorker {
+
+    private final OpenOfficePreferences preferences;
+
+    private final JDialog parent;
+    private boolean foundPaths;
+    private JDialog progressDialog;
+
+    public DetectOpenOfficeInstallation(JDialog parent, OpenOfficePreferences preferences) {
+        this.parent = parent;
+        this.preferences = preferences;
+    }
+
+    public boolean isInstalled() {
+        foundPaths = false;
+        if (preferences.checkAutoDetectedPaths()) {
+            return true;
+        }
+        init();
+        getWorker().run();
+        update();
+        return foundPaths;
+    }
+
+    @Override
+    public void run() {
+        foundPaths = autoDetectPaths();
+    }
+
+    @Override
+    public void init() {
+        progressDialog = showProgressDialog(parent, Localization.lang("Autodetecting paths..."),
+                Localization.lang("Please wait..."));
+    }
+
+    @Override
+    public void update() {
+        progressDialog.dispose();
+    }
+
+    private Optional<Path> selectInstallationPath() {
+        JOptionPane.showMessageDialog(parent,
+                Localization.lang("Unable to autodetect OpenOffice/LibreOffice installation. Please choose the installation directory manually."),
+                Localization.lang("Could not find OpenOffice/LibreOffice installation"),
+                JOptionPane.INFORMATION_MESSAGE);
+        // TODO Dialog service
+        JFileChooser fileChooser = new JFileChooser(new File(System.getenv("ProgramFiles")));
```

Comment:
Can I already use these dialogs for selecting directories only? Looks like they can only be configured as file dialogs.